### PR TITLE
feat: add ui stage link to promo prs

### LINKED ISSUE #1825: Feature: add the Stage link to the Pull Request Promotion created PR
Pull Request Promotion should add the Kargo stage link to the description also.

### PR DESCRIPTION
**Pull request by krancour**
Fixes #1825

### CODE REVIEW

File: internal/controller/promotions/promotions.go [hunk 1]
```diff
@@ -506,13 +507,14 @@ func (r *reconciler) promote(
...
 
 	promoCtx := directives.PromotionContext{
-		WorkDir:         filepath.Join(os.TempDir(), "promotion-"+string(workingPromo.UID)),
-		Project:         stageNamespace,
-		Stage:           stageName,
-		FreightRequests: stage.Spec.RequestedFreight,
-		Freight:         *workingPromo.Status.FreightCollection.DeepCopy(),
-		StartFromStep:   promo.Status.CurrentStep,
-		State:           directives.State(workingPromo.Status.GetState()),
+		APIServerBaseURL: r.cfg.APIServerBaseURL,
```
Comment (hiddeco) [inline internal/controller/promotions/promotions.go:510]: Nit (and non-blocking): this should probably become part of a nested something at some point. As its placement feels odd as-is.

Comment (hiddeco) [reply internal/controller/promotions/promotions.go:510]: Second nit: from the perspective of what we use it for now (within the context of directives), isn't this just the `UIBaseURL`?

Comment (krancour) [reply internal/controller/promotions/promotions.go:510]: Fair point. I'll change that.

Comment (krancour) [reply internal/controller/promotions/promotions.go]: Fixed in fd78be32.

**Review by hiddeco — APPROVED** [review]

**Pull request merged**